### DISCUSSION TITLE
How does server the deployment work in Cloudflare remix vite plugin?

### DISCUSSION THREAD

**Discussion by AdiRishi** [remix-run/remix#8734] [Q&A] [answered]
This might be a stupid question, and I may be missing something obvious, so forgive me 🙏 

I was eyeballing the build output of vite when using remix + cloudflare preset and I can see that we split the build into client and server.
![Screenshot 2024-02-12 at 5 23 34 pm](https://github.com/remix-run/remix/assets/8351234/a5c528a3-8a39-49c4-aee0-94ea78ebe3be)

The docs seem to mention this, and overall it makes sense. But I'm seriously confused on how we deploy the server code?!
The deploy command is `wrangler pages deploy ./build/client`, which seems to skip the server entirely?
Moreover, when I looked into the client directory I didn't see a copy of the server code.

I must be crazy, because I KNOW that this deploy does indeed deploy a worker with the static assets, as I can see the response from a built website is doing SSR. So...how 😕 

One additional question, it was quite nice having control over the server file (traditionally `server.ts`). From the build output it's clear remix uses a default standard server file. In the future, are there plans to re-expose this as an editable file?

**Comment by gustavopch** [comment, marked as answer]
I believe Wrangler will load the server assets from the `functions` folder by default.

**Reply by AdiRishi** [reply]
Oh, you're totally correct! It imports the server build `import * as build from "../build/server";`
Jebus, I totally missed that. Thank you!
